src/Symfony/Bundle/SecurityBundle/DependencyInjection/Security/Factory/RememberMeFactory.php

```diff
@@ -119,34 +119,18 @@ public function getKey()
 
     public function addConfiguration(NodeDefinition $node)
     {
-        $node->fixXmlConfig('user_provider');
         $builder = $node
-            ->beforeNormalization()
-                ->ifTrue(function ($v) { return isset($v['key']); })
-                ->then(function ($v) {
-                    if (isset($v['secret'])) {
-                        throw new \LogicException('Cannot set both key and secret options for remember_me, use only secret instead.');
-                    }
-
-                    @trigger_error('remember_me.key is deprecated since version 2.8 and will be removed in 3.0. Use remember_me.secret instead.', E_USER_DEPRECATED);
-
-                    $v['secret'] = $v['key'];
-
-                    unset($v['key']);
-                })
-                ->end()
-            ->children();
-
-        $builder
-            ->scalarNode('secret')->isRequired()->cannotBeEmpty()->end()
-            ->scalarNode('token_provider')->end()
-            ->arrayNode('user_providers')
-                ->beforeNormalization()
-                    ->ifString()->then(function ($v) { return array($v); })
+            ->fixXmlConfig('user_provider')
+            ->children()
+                ->scalarNode('secret')->isRequired()->cannotBeEmpty()->end()
+                ->scalarNode('token_provider')->end()
+                ->arrayNode('user_providers')
+                    ->beforeNormalization()
+                        ->ifString()->then(function ($v) { return array($v); })
+                    ->end()
+                    ->prototype('scalar')->end()
```

Comment:
This formatting change altering the indentation should be avoided, to make it easier to merge branches together (otherwise we would have to resolve conflicts all the time during the 2.x lifetime when changes are done on this code and propagated to 3.x)